packages/ozone/src/auth-verifier.ts

```diff
@@ -0,0 +1,218 @@
+import express from 'express'
+import * as ui8 from 'uint8arrays'
+import { IdResolver } from '@atproto/identity'
+import { AuthRequiredError, verifyJwt } from '@atproto/xrpc-server'
+
+type ReqCtx = {
+  req: express.Request
+}
+
+type RoleOutput = {
+  credentials: {
+    type: 'role'
+    isAdmin: boolean
+    isModerator: boolean
+    isTriage: true
+  }
+}
+
+type ModeratorOutput = {
+  credentials: {
+    type: 'moderator'
+    aud: string
+    iss: string
+    isAdmin: boolean
+    isModerator: boolean
+    isTriage: true
+  }
+}
+
+type StandardOutput = {
+  credentials: {
+    type: 'standard'
+    aud: string
+    iss: string
+    isAdmin: boolean
+    isModerator: boolean
+    isTriage: boolean
+  }
+}
+
+type NullOutput = {
+  credentials: {
+    type: 'null'
```

Comment:
I struggle a little bit with the string `'null'` here, mostly because I wonder whether it's a typo when I read checks like `credentials.type === 'null'`.

I like keeping a string here so there's a value to discriminate credentials on.  This covers the case where the user did not bring any credentials: would `'none'` work?

Reply:
i like that 👍